Tests/Sentry.Xamarin.Tests/Extensions/IHubExtensionsTests.cs

```diff
@@ -0,0 +1,154 @@
+
+            //Assert
+            logger.DidNotReceive().Log(Arg.Any<SentryLevel>(), Arg.Is<string>(IHubExtensions.DuplicatedBreadcrumbDropped), Arg.Any<Exception>(), Arg.Any<object[]>());
+            Assert.NotEqual(breadcrumb, options.LastInternalBreadcrumb);
+        }
+
+        [Fact]
+        public void AddInternalBreadcrumb_NewBreadcumbWithDifferentType_BreadcrumbAdded()
+        {
+            //Assert
+            const string message = "message";
+            var data = new Dictionary<string, string> { { "key", "value" } };
+            var logger = Substitute.For<IDiagnosticLogger>();
+            var hub = Sut;
+            var options = new SentryXamarinOptions();
+            options.Debug = true;
+            options.DiagnosticLogger = logger;
+            var breadcrumb = new Breadcrumb(message, "type", data);
+            options.LastInternalBreadcrumb = breadcrumb;
+            
+            //Act
+            hub.AddInternalBreadcrumb(options, message, null, "type2", data);
+
+            //Assert
+            logger.DidNotReceive().Log(Arg.Any<SentryLevel>(), Arg.Is<string>(IHubExtensions.DuplicatedBreadcrumbDropped), Arg.Any<Exception>(), Arg.Any<object[]>());
+            Assert.NotEqual(breadcrumb, options.LastInternalBreadcrumb);
+        }
+
+        [Fact]
+        public void AddInternalBreadcrumb_NewBreadcumbWithDifferentData_BreadcrumbAdded()
+        {
+            //Assert
+            const string message = "message";
+            const string type = "type";
+            var logger = Substitute.For<IDiagnosticLogger>();
+            var hub = Sut;
+            var options = new SentryXamarinOptions();
+            options.Debug = true;
+            options.DiagnosticLogger = logger;
+            var breadcrumb = new Breadcrumb(message, type, new Dictionary<string, string> { { "key", "value" } });
+            options.LastInternalBreadcrumb = breadcrumb;
+
+            //Act
+            hub.AddInternalBreadcrumb(options, message, null, type, new Dictionary<string, string> { { "key", "value2" } });
+
+            //Assert
+            logger.DidNotReceive().Log(Arg.Any<SentryLevel>(), Arg.Is<string>(IHubExtensions.DuplicatedBreadcrumbDropped), Arg.Any<Exception>(), Arg.Any<object[]>());
+            Assert.NotEqual(breadcrumb, options.LastInternalBreadcrumb);
+        }
+
+        [Fact]
+        public async Task AddInternalBreadcrumb_DuplicatedIs2SecondsAhead_BreadcrumbAdded()
+        {
+            //Assert
+            const string message = "message";
+            const string type = "type";
+            var data = new Dictionary<string, string> { { "key", "value" } };
+            var logger = Substitute.For<IDiagnosticLogger>();
+            var hub = Sut;
+            var options = new SentryXamarinOptions();
+            options.Debug = true;
+            options.DiagnosticLogger = logger;
+
+            var breadcrumb = new Breadcrumb(message, type, data);
+            await Task.Delay(3000);
```

Comment:
Tests need to avoid these sleeps. This test will take at least 3 seonds, that's quite a lot.
Ideally you'd mock things so you could "pretend" it took 3 seconds and still validate your logic.